web/packages/design/src/Tooltip/HoverTooltip.tsx

```diff
@@ -143,38 +182,26 @@ export const HoverTooltip = ({
     useRole(context, { role: 'tooltip' }),
   ]);
 
-  if (!tipContent) {
+  if (!tipContent || !children) {
     return <>{children}</>;
   }
 
-  const handleMouseEnter = (event: React.MouseEvent<Element>) => {
-    const { currentTarget } = event;
-    contentRef.current = currentTarget as HTMLElement;
-
-    if (showOnlyOnOverflow) {
-      if (
-        currentTarget instanceof Element &&
-        currentTarget.parentElement &&
-        currentTarget.scrollWidth > currentTarget.parentElement.offsetWidth
-      ) {
-        setOpen(true);
-      }
-      return;
-    }
-
-    setOpen(true);
-  };
+  let childWithRef: ReactElement<{ ref: React.Ref<HTMLElement> }> | undefined;
+  try {
+    // The type of `children` is `ReactElement` which allows only one child.
+    childWithRef = Children.only(children);
+  } catch (e) {
+    logger.error('Provided invalid child', e);
```

Comment:
Why do we swallow this error? If it's invalid, let's just allow it to propagate. If we tolerate this, then it's valid by definition. Perhaps in this case we should assert that `showOnlyOnOverflow` is set to `false` in this case?